### PR TITLE
Fix Delta INSERT when schema OR table location ends with two slashes

### PR DESCRIPTION
**Pull request by findepi**
Extracted from https://github.com/trinodb/trino/pull/17958

For https://github.com/trinodb/trino/issues/17966, but does not fix it fully.

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java [hunk 1]
```diff
@@ -115,79 +115,80 @@ protected void validateFilesAfterOptimize(String location, Set<String> initialFi
...
+        try (UncheckedCloseable ignoredDropSchema = onClose("DROP SCHEMA " + schemaName)) {
+            assertThat(getSchemaLocation(schemaName)).isEqualTo(schemaLocation);
 
-        assertUpdate("INSERT INTO " + qualifiedTableName + "  VALUES ('str1', 1), ('str2', 2), ('str3', 3)", 3);
-        assertQuery("SELECT * FROM " + qualifiedTableName, "VALUES ('str1', 1), ('str2', 2), ('str3', 3)");
+            assertUpdate("CREATE TABLE " + qualifiedTableName + "(col_str varchar, col_int int)" + partitionQueryPart);
+            try (UncheckedCloseable ignoredDropTable = onClose("DROP TABLE " + qualifiedTableName)) {
+                String expectedTableLocation = ((schemaLocation.endsWith("/") ? schemaLocation : schemaLocation + "/") + tableName)
+                        // Hive normalizes double slash
+                        .replaceAll("(?<!(s3:))/+", "/");
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveS3AndGlueMetastoreTest.java:165]: nit: double slash → multiple slash?

**Review by ebyhr — APPROVED** [review]
Looks good except for CI failures.

**Comment by findinpath** [comment]
Build is :red_circle: 

```
Error:  io.trino.plugin.deltalake.metastore.glue.TestDeltaS3AndGlueMetastoreTest.testBasicOperationsWithProvidedTableLocation[true, TWO_TRAILING_SLASHES](3)  Time elapsed: 2.974 s  <<< FAILURE!
java.lang.AssertionError: 

Expecting code to raise a throwable.
	at io.trino.plugin.hive.BaseS3AndGlueMetastoreTest.testBasicOperationsWithProvidedTableLocation(BaseS3AndGlueMetastoreTest.java:110)
```

```
Error:  io.trino.plugin.deltalake.metastore.glue.TestDeltaS3AndGlueMetastoreTest.testBasicOperationsWithProvidedTableLocation[true, TWO_TRAILING_SLASHES](3)  Time elapsed: 2.974 s  <<< FAILURE!
java.lang.AssertionError: 

Expecting code to raise a throwable.
	at io.trino.plugin.hive.BaseS3AndGlueMetastoreTest.testBasicOperationsWithProvidedTableLocation(BaseS3AndGlueMetastoreTest.java:110)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

**Review by findinpath — APPROVED** [review]

**Comment by findepi** [comment]
fixed iceberg test failures, but didn't fix delta yet

**Comment by findepi** [comment]
CI `ci / pt (default, suite-8-non-generic, )` https://github.com/trinodb/trino/issues/15187

**Comment by findepi** [comment]
CI [pt (hdp3, suite-1, )](https://github.com/trinodb/trino/actions/runs/5332425529/jobs/9662369164?pr=17980#logs) https://github.com/trinodb/trino/issues/16315

**Pull request merged**